pontoon/base/models.py

```diff
@@ -443,7 +449,13 @@ def get_latest_activity(self, locale=None):
             else:
                 return None
         else:
-            return ProjectLocale.get_latest_activity(self, locale)
+            if hasattr(self, 'fetched_project_locales'):
```

Comment:
I would refactor get_latest_activity into some kind of mixin, to avoid unecesssary code duplication.